### PR TITLE
Prepare the 1.6.4 release.

### LINKED ISSUE #681: Release 1.6.4
On the docket:
+ [x] Restore pex.pex_bootstrapper.is_compressed API #684
+ [ ] Release more flexible pex binaries. #654
  + [x] If an `--interpreter-constraint` is set, it should always be honored. #656

### PR DESCRIPTION
**Pull request by jsirois**
Fixes #681

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
+* Add the version of pex used to build a pex to build_properties. (#687)
+  `PR #687 <https://github.com/pantsbuild/pex/pull/687>`_
+
+* Honor interpreter constraints even when PEX_PYTHON and PEX_PYTHON_PATH not set (#668)
+  `PR #668 <https://github.com/pantsbuild/pex/pull/668>`_
+
 1.6.3
 -----
 
+This release changes the behavior of the ``--interpreter-constraint`` option.
```
Comment (jsirois) [inline CHANGES.rst]: I missed this warning when releasing 1.6.3. Backfilling here and already updated on the GitHub release: https://github.com/pantsbuild/pex/releases/tag/v1.6.3

File: CHANGES.rst [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+Previously, interpreter constraints were ANDed which made it impossible to
```
Comment (Eric-Arellano) [inline CHANGES.rst]: Comma needed after ANDed.

File: CHANGES.rst [hunk 3]
```diff
@@ -1,9 +1,31 @@
...
+express constraints like '>=2.7,<3' OR '>=3.6,<4'; ie: either python 2.7 or
+else any python 3 release at or above 3.6. Now interpreter constraints are
+ORed which is likely a breaking change if you have scripts that pass multiple
```
Comment (Eric-Arellano) [inline CHANGES.rst]: Comma needed before which

File: CHANGES.rst [hunk 4]
```diff
@@ -1,9 +1,31 @@
...
+interpreter constraints. To transition, use the native ``,`` AND operator in
+your constraint expression as used in the example above.
```
Comment (Eric-Arellano) [inline CHANGES.rst]: Comma needed before as.

**Comment by jsirois** [comment]
Thanks Eric - fixed those up.

**Pull request merged**